xarray/backends/core.py

```diff
@@ -0,0 +1,425 @@
+from collections import Mapping, defaultdict
+import copy
+import os.path
+import numpy as np
+import pickle
+import threading
+from xarray.core.pycompat import OrderedDict, dask_array_type
+
+from xarray import Variable
+
```

Comment:
nit: you need to import `xarray`, and `conventions`. I'm guessing you are also expecting to put all the Coders in a module `coders`?